pipes.js

```diff
@@ -4,47 +4,56 @@
+ *  myPipe.on("default","before",function (options) {
+ *   return options;
+ *  });
+ *
+ *  myPipe.on("default","action",function (options) {
+ *   return options;
+ *  });
+ *  ...
+ *  ```
  *  "before" actions will be executed 1st, "action" 2nd and "after" 3rd.
- *  See the docs for Pipes.prototype.on on how to attach actions to pipelines. 
+ *  See the docs for Pipes.prototype.on on how to attach actions to pipes. 
  */
 Pipes = function () {
-  var pipelines = _.flatten(_.toArray(arguments))
+  var stages = _.flatten(_.toArray(arguments))
 
-  var isEmpty = (pipelines.length < 1);
-  var pipelineNamesAreNotStrings = !! _.find(pipelines, function (pipeline) { return ! _.isString(pipeline); });
-  var pipelineNamesAreNotUnique = !! _.find(pipelines, function(pipeline, index, allPipelines) {
-    var nameCount = _.filter(allPipelines, function (stg) { return stg === pipeline; });
+  var isEmpty = (stages.length < 1);
+  var stageNamesAreNotStrings = !! _.find(stages, function (stage) { return ! _.isString(stage); });
+  var stageNamesAreNotUnique = !! _.find(stages, function(stage, index, allPipelines) {
+    var nameCount = _.filter(allPipelines, function (stg) { return stg === stage; });
     return nameCount.length != 1;
   });
 
-  if ( isEmpty && pipelineNamesAreNotStrings ) {
-    throw new Meteor.Error("invalid-arguments","pipelineNames argument does not exist or is not an Array");
-  } else if ( pipelineNamesAreNotUnique ) {
-    throw new Meteor.Error("duplicate-pipelinenames","pipelineNames must be unique");
+  if ( isEmpty && stageNamesAreNotStrings ) {
+    throw new Meteor.Error("invalid-arguments","stageNames argument does not exist or is not an Array");
+  } else if ( stageNamesAreNotUnique ) {
+    throw new Meteor.Error("duplicate-stagenames","stageNames must be unique");
   } else {
-    this.pipelines = pipelines;
+    this._stages = stages;
   }
 
   this.actions = {};
-
-  var __actions = this.actions;
-  _.each(pipelines, function(pipeline) {
-    __actions[pipeline] = [];
-  });
 };
 
 /**
- * Adds a new action to the predefined pipeline pipeline.
+ * Adds a new action to one of the predefined stages for a pipe.
  *
  * @method  Pipes.prototype.on
- * @param {string} pipelineName The name of the pipeline
- * @param {function} action The new action. Each new action is added at the end of the pipeline i.e. will be executed last in the specific pipeline sequence.
+ * @param {string} pipeName The name of the pipe
+ * @param {string} stageName The name of the stage. This affects the order of execution. See docs on Pipes constructor for more.  
+ * @param {function} action The new action. Each new action is added at the end of the stage i.e. will be executed last in the 
+ * specific stage sequence for the specified pipe.
  *
  */
- Pipes.prototype.on = function (pipelineName, action) {
+ Pipes.prototype.on = function (pipeName, stageName, action) {
+  var self = this;
   if (_.isString(action)) {
```

Comment:
The option to specify an action as a string is untested, because `Pipes.prototype.do` takes two arguments, and you here only specify a single argument, it won't work.